Common/Source/Logger/NMEAlogger.cpp

```diff
@@ -64,36 +64,26 @@ static bool wasWriting=false;
   if(iLastPort != -1)  /* already a port info ? */
   {
     if( iLastPort != PortNum) /* more than one port active (another than the previous) */
+    {
+      if((PortNum >=0) && (PortNum < NUMDEV))
+      {
+        if(logfsingle[PortNum] == NULL)
 	{
-	  if(logfp0 == NULL)
-	  {
-                TCHAR fpname[LKSIZEBUFFERPATH];
-                TCHAR buffer[LKSIZEBUFFERPATH];
-		LocalPath(buffer,TEXT(LKD_LOGS));
-	    _stprintf(fpname, _T("%s%sNMEA_A_%04d-%02d-%02d-%02d-%02d-%02d.txt"), buffer, _T(DIRSEP), GPS_INFO.Year, GPS_INFO.Month, GPS_INFO.Day, GPS_INFO.Hour, GPS_INFO.Minute, GPS_INFO.Second);
-		logfp0 = _tfopen(fpname, _T("a"));
-		if (logfp0 == NULL) {
-		  DoStatusMessage(_T("CANNOT SAVE TO NMEA LOGFILE PORT A:"));
-		  return; }
-                wasWriting=true;
-	  }
-
-	  if(logfp1 == NULL)
-	  {
-                TCHAR fpname[LKSIZEBUFFERPATH];
-                TCHAR buffer[LKSIZEBUFFERPATH];
-        LocalPath(buffer,TEXT(LKD_LOGS));
-		_stprintf(fpname, _T("%s%sNMEA_B_%04d-%02d-%02d-%02d-%02d-%02d.txt"), buffer, _T(DIRSEP), GPS_INFO.Year, GPS_INFO.Month, GPS_INFO.Day, GPS_INFO.Hour, GPS_INFO.Minute, GPS_INFO.Second);
-		logfp1 = _tfopen(fpname, _T("a"));
-		if (logfp1 == NULL) {
-		  DoStatusMessage(_T("CANNOT SAVE TO NMEA LOGFILE PORT B:"));
-		  return; }
-                wasWriting=true;
-	  }
+          TCHAR fpname[LKSIZEBUFFERPATH];
+          TCHAR buffer[LKSIZEBUFFERPATH];
+	  LocalPath(buffer,TEXT(LKD_LOGS));
+	  _stprintf(fpname, _T("%s%sNMEA_%C_%04d-%02d-%02d-%02d-%02d-%02d.txt"), buffer, _T(DIRSEP),'A'+PortNum, GPS_INFO.Year, GPS_INFO.Month, GPS_INFO.Day, GPS_INFO.Hour, GPS_INFO.Minute, GPS_INFO.Second);
```

Comment:
error: format specifies type 'wchar_t' but the argument has type 'int' [-Werror,-Wformat]
     [ "%C" <-> 'A'+PortNum ] need to be replaced by  [ "%c" <-> _T('A')+PortNum ]